src/Serializer/SnippetDenormalizer.php

```diff
@@ -30,16 +30,16 @@ public function denormalizeSnippet(array $item) : PromiseInterface
         $this->identityMap->reset($id);
 
         if (empty($this->globalCallback)) {
-            $this->globalCallback = new CallbackPromise(function () {
+            $this->globalCallback = new Promise(function () {
                 $this->identityMap->fillMissingWith($this->fetchComplete);
 
-                $this->globalCallback = null;
-
                 $settled = $this->identityMap->waitForAll();
 
                 $this->identityMap->reset();
 
-                return $settled;
+                $this->globalCallback->resolve($settled);
```

Comment:
I like simplification.
Is it correct to say that `$settled` end up only being used for the ids that are denormalized? All models promises share the same `globalCallback` and access their own id from it.